src/Shared/Microsoft.DotNet.GitHub.Authentication/GitHubAppTokenProvider.cs

```diff
@@ -2,55 +2,53 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-using GitHubJwt;
+using System;
+using System.IdentityModel.Tokens.Jwt;
+using System.Security.Cryptography;
 using Microsoft.Extensions.Options;
 using System.Text;
+using Microsoft.Extensions.Internal;
+using Microsoft.IdentityModel.Tokens;
 
 namespace Microsoft.DotNet.GitHub.Authentication
 {
     public class GitHubAppTokenProvider : IGitHubAppTokenProvider
     {
+        private readonly ISystemClock _clock;
         private readonly IOptionsMonitor<GitHubTokenProviderOptions> _options;
 
-        public GitHubAppTokenProvider(IOptionsMonitor<GitHubTokenProviderOptions> options = null)
+        public GitHubAppTokenProvider(ISystemClock clock, IOptionsMonitor<GitHubTokenProviderOptions> options = null)
         {
+            _clock = clock;
             _options = options;
         }
 
         public string GetAppToken()
         {
-            return GetAppToken(Options.DefaultName);
+            var options = _options.CurrentValue;
+            return GetAppToken(options.GitHubAppId, options.PrivateKey);
         }
-
-        /// <summary>
-        /// Get an app token using the <see cref="GitHubTokenProviderOptions"/> corresponding to the specified <see href="https://docs.microsoft.com/en-us/dotnet/core/extensions/options#named-options-support-using-iconfigurenamedoptions">named option</see>.
-        /// </summary>
```

Comment:
Comment is still valid, why remove it?

Reply:
I didn't mean to.  There was a HUGE merge conflict in this file, and I must have resolved it badly. I'll put it back. :-)

Reply:
(Though it's also a little useless to comment the provider, and not the service, because no one ever has a reference to the provider, only the service, so the comments aren't really visible... and it doesn't matter what _this_ class says it's does, only what the interface is designed to do. :-)

Reply:
Put it back. :-)

Reply:
Yeah, I could have been more thorough in adding the comments. But every bit helps, right? 😄